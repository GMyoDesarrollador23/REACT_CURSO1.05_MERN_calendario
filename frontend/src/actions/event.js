import Swal from "sweetalert2";

import { types } from "../types/types";
import { fetchConToken } from "../helpers/fetch";
import { prepereevents } from "../helpers/prepereEvents";

// resiveir el evento activo
export const eventSetActive = (event) => ({
   type: types.eventSetActive,
   payload: event,
});

// limpiar nota activa
export const eventClearActiveNote = () => ({
   type: types.eventClearActiveEvent,
});

// purgar la nota activa cuado el usuario cierra cesion
export const eventLogout = () => ({
   type: eventLogout,
});


// POST -> CREATE
export const eventStartAddNew = (event) => {
   return async (dispatch, getState) => {
      const { uid, name } = getState().auth;

      try {
         // enviar el evento al backend
         const resp = await fetchConToken(
            "events",
            event,
            "POST"
         );
         const body = await resp.json();

         if (body.ok) {
            // aniadir el id y el usuario
            event.id = body.evento.id;
            event.user = {
               _id: uid,
               name,
            };

            // dispara el evento
            dispatch(eventAddNew(event));
         }
      } catch (error) {
         console.error(error);
      }
   };
};

// GET -> READ
export const eventStartLoading = () => {
   return async (dipatch) => {
      try {
         // recuperando los eventos de la base de datos
         const resp = await fetchConToken("events");
         const body = await resp.json();
         const events = prepereevents(body.eventos);

         // console.log(events);

         dipatch(eventLoaded(events));
      } catch (error) {
         console.error(error);
      }
   };
};

// PUT -> UPDATE
export const eventsStartUpdate = (event) => {
   return async (dispatch) => {
      try {
         // console.log(event);
         const resp = await fetchConToken(
            `events/${event.id}`,
            event,
            "PUT"
         );
         const body = await resp.json();
         if (body.ok) {
            dispatch(eventUpdate(event));
         } else {
            Swal.fire("Error", body.msg, "error");
         }
      } catch (error) {
         console.error(error);
      }
   };
};

// DELETE
export const eventsStartDelete = () => {
   return async (dispatch, getState) => {
      const { activeEvent } = getState().calendar;
      try {
         const resp = await fetchConToken(
            `events/${activeEvent.id}`,
            {},
            "DELETE"
         );
         const body = await resp.json();
         if (body.ok) {
            dispatch(eventDeleted());
         } else {
            Swal.fire("Error", body.msg, "error");
         }
      } catch (error) {
         console.error(error);
      }
   };
};
// ---------------------------------------------------------

// aniadir nuevo evento
const eventAddNew = (event) => ({
   type: types.eventAddNew,
   payload: event,
});

// cargar Eventos
const eventLoaded = (events) => ({
   type: types.eventLoaded,
   payload: events,
});

// actualizar nota activa
const eventUpdate = (event) => ({
   type: types.eventUpdated,
   payload: event,
});

// eliminar nota activa
const eventDeleted = () => ({
   type: types.eventDeleted,
});
