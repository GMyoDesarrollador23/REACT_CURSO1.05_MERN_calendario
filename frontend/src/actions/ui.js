import { types } from "../types/types";

// abrir modal
export const uiOpenModal = () => ({
   type: types.uiOpenModal,
});

// cerrar modal
export const uiCleseModal = () => ({
   type: types.uiCleseModal,
});
