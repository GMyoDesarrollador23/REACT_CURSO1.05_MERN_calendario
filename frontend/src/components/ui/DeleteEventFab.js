import React from "react";
import { useDispatch } from "react-redux";
import { eventsStartDelete } from "../../actions/event";

export const DeleteEventFab = () => {
   const dispatch = useDispatch();

   const handleDelete = () => {
      dispatch(eventsStartDelete());
   };

   return (
      <button
         className="btn btn-outline-danger fab-danger"
         onClick={handleDelete}
      >
         <i className="fas fa-trash"></i>
         <span> Eliminar Evento</span>
      </button>
   );
};
