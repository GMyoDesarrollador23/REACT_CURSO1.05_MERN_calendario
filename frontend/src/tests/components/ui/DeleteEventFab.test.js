import React from "react";
import { mount } from "enzyme";
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import { DeleteEventFab } from "../../../components/ui/DeleteEventFab";
import { eventsStartDelete } from "../../../actions/event";

// *******************************************************
jest.mock("../../../actions/event", () => ({
   eventsStartDelete: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {};
let store = mockStore(initialState);
store.dispatch = jest.fn();

const wrapper = mount(
   <Provider store={store}>
      <DeleteEventFab />
   </Provider>
);

// ******************************************************
describe("pruebas en el <DeleteEventFab/>", () => {
   // --------------------------------------------
   test("debe de mostrarse correctamente", () => {
      expect(wrapper).toMatchSnapshot();
   });
   // --------------------------------------------

   test("debe de hecer la llamada", () => {
      wrapper.find("button").prop("onClick")();
      expect(eventsStartDelete).toHaveBeenCalled();
   });
   // --------------------------------------------
});
