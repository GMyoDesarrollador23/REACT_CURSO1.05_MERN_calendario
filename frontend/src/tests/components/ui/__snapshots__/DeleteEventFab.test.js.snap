// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <DeleteEventFab/> debe de mostrarse correctamente 1`] = `
<button
  className="btn btn-outline-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Eliminar Evento
  </span>
</button>
`;
